src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -85,172 +89,299 @@ internal class SqlServerFhirDataStore : IFhirDataStore, IProvideCapability
+                            if (!resourceExt.AllowCreate)
                             {
-                                throw new PreconditionFailedException(string.Format(Core.Resources.IfMatchHeaderRequiredForResource, resource.ResourceTypeName));
+                                throw new MethodNotAllowedException(Core.Resources.ResourceCreationNotAllowed);
                             }
 
-                            throw new BadRequestException(string.Format(Core.Resources.IfMatchHeaderRequiredForResource, resource.ResourceTypeName));
+                            resource.Version = InitialVersion;
                         }
-
-                        if (resource.IsDeleted && existingResource.IsDeleted)
+                        else
                         {
-                            // Already deleted - don't create a new version
-                            return null;
-                        }
+                            if (resourceExt.RequireETagOnUpdate && !eTag.HasValue)
+                            {
+                                // This is a versioned update and no version was specified
+                                // TODO: Add this to SQL error codes in AB#88286
+                                // The backwards compatibility behavior of Stu3 is to return 412 Precondition Failed instead of a 400 Bad Request
+                                if (_modelInfoProvider.Version == FhirSpecification.Stu3)
+                                {
+                                    throw new PreconditionFailedException(string.Format(Core.Resources.IfMatchHeaderRequiredForResource, resource.ResourceTypeName));
+                                }
+
+                                throw new BadRequestException(string.Format(Core.Resources.IfMatchHeaderRequiredForResource, resource.ResourceTypeName));
+                            }
 
-                        // check if reosurces are equal if its not a Delete action
-                        if (!resource.IsDeleted)
-                        {
-                            // check if the new resource data is same as existing resource data
-                            if (string.Equals(RemoveVersionIdAndLastUpdatedFromMeta(existingResource), RemoveVersionIdAndLastUpdatedFromMeta(resource), StringComparison.Ordinal))
+                            if (resource.IsDeleted && existingResource.IsDeleted)
                             {
-                                // Send the existing resource in the response
-                                return new UpsertOutcome(existingResource, SaveOutcomeType.Updated);
+                                // Already deleted - don't create a new version
+                                results.Add(resourceKey, null);
+                                continue;
                             }
+
+                            // check if reosurces are equal if its not a Delete action
```

Comment:
```suggestion
                            // check if resources are equal if its not a Delete action
```

Reply:
I will remove garbage copy/pasted from previous code as is.